### ISSUE TITLE
NetworkIF add support for additional fields

### ISSUE THREAD

**Issue by spyhunter99** [oshi/oshi#696]
would be awesome if we could support the same fields that sigar supported

- [x] Rx dropped packets
- [ ] RX frames
- [ ] RX overruns
- [ ] TX Carrier
- [x] TX collisions
- [ ] TX overruns
- [ ] uptime (time since the nic was last administratively enabled or system boot time
- [ ] connected time, time elapsed since the nic was connected to a network or 0 if it is currently not connected
- [x] NIC subnet mask

**Comment by dbwiddis** [comment]
Would be awesome if you could submit a PR for that!

**Comment by dbwiddis** [comment]
Slightly more serious reply... good idea but before we get into the same trap we did with `OSProcess` data and using lots of CPU, this may be something that might wait until we have a better handle on "don't grab info until you need it" per #548

**Comment by dbwiddis** [comment]
If someone's working on this, see #976 for another potential addition.

**Comment by dbwiddis** [comment]
I took a look at doing this myself, and am not sure why the specific statistics shown were selected from a lot more alterantives (Linux) and the data is sparse on other OS's.   Some feedback:
 - Windows does not seem to provide any of this information.  The closest it gets is "Discards" which would probably be highly correlated to "Dropped Packets"
 - It does look like we could use our existing queries to add "rx_dropped" and "collisions" on most of the *nix-based OS's
 - The other fields (frames, overruns, carrier) appear to be subsets of the errors already reported.  There are many more error types on Linux, but no more detail on other OS's.  
 - I didn't see how to get uptime/connected time on any OS.  It's probably there somewhere, probably in a command line call.   Do you have links to the SIGAR source for those that I can look at to see if it's easily accessible?

Bottom line:  I'll probably work on adding dropped (using rx_discards for Windows) and collisions (using tx_discards for Windows) for the 4.4.0 release, and look more into how easy timing data is to get; but other than that, I don't think there's much value in adding only a select few of the many Linux error details, which are accessible via `getLongFromFile()` and similar methods in the util class, for those who want to automate them.

**Comment by spyhunter99** [comment]
for windows, there is a wmi query for network connected time and i think there's another timestamp for last reset time

**Comment by dbwiddis** [comment]
I believe you're referring to [Win32_NetworkAdapter](https://docs.microsoft.com/en-us/windows/win32/cimwin32prov/win32-networkadapter)?  (I was going to try to use that to test for Physical adapters as well, although since all the other info is available in a native call I hesitate to introduce a WMI call and its associated latency for a single field.) I see the `TimeOfLastReset` field but nothing about network connected time.

**Comment by spyhunter99** [comment]
it must have only been `TimeOfLastReset`  which appears to be uptime as defined above (maybe a better label is needed)

**Comment by dbwiddis** [comment]
Closing this as #1101 added dropped packets and collisions, the remaining "standard" fields in `netstat` and cross-platform friends.  

I don't plan on adding any of the other choices, and this issue has been open long enough for others with interest to contribute.  The remaining error details appear to be Linux-only and are only a portion of the available results.  It's relatively easy to fetch the data should a user wish to use the existing Linux source as inspiration.  The restart time appears to be Windows only and does not appear to be available at all in *nix systems.
